analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -25,9 +25,10 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
     public sealed record SymbolicValue
     {
         // Reuse instances to save memory. This "True" has the same semantic meaning and any other symbolic value with BoolConstraint.True constraint
-        public static readonly SymbolicValue This = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
-        public static readonly SymbolicValue Null = new SymbolicValue().WithConstraint(ObjectConstraint.Null);
-        public static readonly SymbolicValue NotNull = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
+        public static readonly SymbolicValue Constraintless = new();
+        public static readonly SymbolicValue This = Constraintless.WithConstraint(ObjectConstraint.NotNull);
```

Comment:
Probably for another PR: Seeing this, I don't think we'll need this as a separate field (it was in the old engine). By removing it, we could remove `InstanceReference` operation processor and use `NotNullOperation` processor instead.
```suggestion
```

Reply:
Let's do it another PR.

Reply:
Tracked by #6964